src/dbSta/src/dbReadVerilog.cc

```diff
@@ -167,6 +189,14 @@ void dbLinkDesign(const char* top_cell_name,
       top_cell_name, link_make_black_boxes, verilog_network->report());
   if (success) {
     Verilog2db v2db(verilog_network, db, logger);
+    Tcl_Interp* tcl_interp = OpenRoad::openRoad()->tclInterp();
```

Comment:
I would personally expose some tcl command to modify v2db.heirachy instead of trying to run TCL here. Then code that's testing this code will run when the flag is set.

Reply:
To help, the link command has been expanded to include -hier as an option. To run a test look at the test case readdb_hier.tcl.

set_debug_level ODB  "dbReadVerilog" 1
link_design top -hier